x-pack/plugins/observability_solution/inventory/public/components/grouped_inventory/grouped_grid_wrapper.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiDataGridSorting } from '@elastic/eui';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
+import { useKibana } from '../../hooks/use_kibana';
+import { EntitiesGrid } from '../entities_grid';
+import type { EntityColumnIds, EntityType } from '../../../common/entities';
+import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import { useInventoryParams } from '../../hooks/use_inventory_params';
+import { useInventoryRouter } from '../../hooks/use_inventory_router';
+import { useInventoryState } from '../../hooks/use_inventory_state';
+
+interface Props {
+  field: string;
+}
+
+export function GroupedGridWrapper({ field }: Props) {
+  const { pagination, setPagination } = useInventoryState();
+  const { query } = useInventoryParams('/');
+  const { sortField, sortDirection, kuery } = query;
+  const inventoryRoute = useInventoryRouter();
+  const pageIndex = (field ? pagination?.[field] : pagination?.unified) ?? 0;
```

Comment:
- Will `field` ever be undefined or empty at this stage?
- Does it make sense to call `pagination?.unified` on this component?